### PR TITLE
Add Python 3 support to Artman

### PR DESCRIPTION
**Pull request by lukesneeringer**
@ethanbao I was going to do some UX hacking on Artman, and discovered that it does not support Python 3. This pull request fixes that.

It does the following:

  * Updates the code to support Python 2.7 and 3.5. This mostly means `print` is consistently used as a function now, and calls to `urllib`, `urlparse`, and `urllib2` now all pull from `six.moves.urllib`, which mirrors the Python 3 package.
  * Substantially refactors the test running. Running tox now does the following, in order:
      * Runs the Python 2.7 tests.
      * Runs the Python 3.5 tests.
      * Runs the linter (on Python 3.5).
      * Puts the coverage data together and produces a report.
  * The `.travis.yml` is also changed to reflect this.

### CODE REVIEW

File: execute_pipeline.py [hunk 1]
```diff
@@ -34,20 +34,22 @@
...
 
 """
-
-import sys
+from __future__ import print_function
 import argparse
 import ast
 import base64
 import os
+import sys
```
Comment (ethanbao) [inline execute_pipeline.py]: What is this import for?

Comment (lukesneeringer) [inline execute_pipeline.py]: I just moved it down from a few lines above so it would be in alphabetical order.

**Comment by ethanbao** [comment]
LGTM, thanks for the refactoring!

**Comment by lukesneeringer** [comment]
There is an intermittent failure on one of the tests on Python 3.4 and Python 3.5 only. I am trying to figure out what the issue is.

**Comment by lukesneeringer** [comment]
The intermittent failure is that the test is depending on a dictionary serialization when dictionaries are unordered. Fixing.

**Pull request merged**